appservice/src/main/java/nu/yona/server/subscriptions/rest/UserPhotoController.java

```diff
@@ -142,10 +151,10 @@ protected UserPhotoResource instantiateResource(UserPhotoDto userPhoto)
 			return new UserPhotoResource(userPhoto);
 		}
 
-		private void addSelfLink(Resource<UserPhotoDto> userPhotoResource)
+		private void addPhotoLink(Resource<UserPhotoDto> userPhotoResource)
 		{
-			userPhotoResource
-					.add(UserPhotoController.getUserPhotoLinkBuilder(userPhotoResource.getContent().getId()).withSelfRel());
+			userPhotoResource.add(UserPhotoController.getUserPhotoLinkBuilder(userPhotoResource.getContent().getId().get())
+					.withRel("yona:userPhoto"));
```

Comment:
The relationship should be "userPhoto". The curie is added by the HAL serialization.